src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -25,7 +26,37 @@ public partial class DependencyObjectCollectionBase : DependencyObject
 	public partial class DependencyObjectCollection<T> : DependencyObjectCollectionBase, IList<T>, IEnumerable<T>, IEnumerable, IObservableVector<T>
 		where T : DependencyObject
 	{
-		public event VectorChangedEventHandler<T> VectorChanged;
+		private object _vectorChangedHandlersLock = new();
+
+		// Explicit handlers list to avoid the cost of generic multicast
+		// delegates handling on mono's AOT.
+		private List<VectorChangedEventHandler<T>> _vectorChangedHandlers;
+
+		public event VectorChangedEventHandler<T> VectorChanged
+		{
+			add
+			{
+				lock (_vectorChangedHandlersLock)
```

Comment:
I would assume we don't need a `lock` here since it's about `DependencyObject` the event should be touched only from the UI thread?

Reply:
They may be registered from a background thread, even if unlikely. It's there to keep the original behavior.